src/nvim/api/vim.c

```diff
@@ -778,7 +778,21 @@ void nvim_echo(ArrayOf(Tuple(String, *HLGroupID)) chunks, Boolean history, Dict(
     kind = opts->err ? "echoerr" : history ? "echomsg" : "echo";
   }
 
-  msg_multihl(hl_msg, kind, history, opts->err);
+  String status = opts->status;
+  if ((kind != NULL && strcmp(kind, "progress") == 0)
+      && ((status.data == NULL)
+          || (strcmp(status.data, "success") != 0
```

Comment:
What would be good candidate options for status? also if status is failed should we mark it as error message?